src/main/java/seedu/address/model/applicant/Nric.java

```diff
@@ -8,13 +8,13 @@
  */
 public class Nric {
 
-    public static final String MESSAGE_CONSTRAINTS = "Nric can take any values, and it should not be blank";
+    public static final String MESSAGE_CONSTRAINTS = "Nric must be a string with the format "
+            + "[Char][7 numeric digits][Char]";
 
     /*
-     * The first character of the Nric must not be a whitespace,
-     * otherwise " " (a blank string) becomes a valid input.
+     * The Nric must be in the format [Char][7 numeric digits][Char].
      */
-    public static final String VALIDATION_REGEX = "[^\\s].*";
+    public static final String VALIDATION_REGEX = "^[a-zA-z]\\d{7}[a-zA-Z]$";
```

Comment:
One suggestion will be to change the first part of regex [a-zA-z] to [a-zA-Z] as well just to be consistent.
(though I don't think there should be any issues performance-wise since A-z ranges are bigger than A-Z)